hudi-timeline-service/src/main/java/org/apache/hudi/timeline/service/RequestHandler.java

```diff
@@ -187,12 +200,20 @@ private boolean syncIfLocalViewBehind(Context ctx) {
       SyncableFileSystemView view = viewManager.getFileSystemView(basePath);
       synchronized (view) {
         if (isLocalViewBehind(ctx)) {
-          HoodieTimeline localTimeline = viewManager.getFileSystemView(basePath).getTimeline();
-          LOG.info("Syncing view as client passed last known instant " + lastKnownInstantFromClient
-              + " as last known instant but server has the following last instant on timeline :"
-              + localTimeline.lastInstant());
-          view.sync();
-          return true;
+          try {
```

Comment:
All the code block is thread safe already because of the `synchronized (view)`.

Reply:
cc @nbalajee Do you have interest for the reviewing?

Reply:
As @nsivabalan mentioned, this is not thread-safe before, as some requests may completely skip the synchronized block as `isLocalViewBehind` returns false while the `view.sync()` is still in progress, and the read-write lock does not prevent the stale file system view to be served.

Reply:
Did you mean when the local view is  not behind, it still needs to wait for the next refresh? Sorry I didn’t think that is the correct direction to go. 

And there is another question you didn’t answer, why it is okey when MDT is disabled? They share the same lock guard right?

Reply:
yeah this seems like a good use of read/write locks.. but then, the whole thing can be inside a synchronized block for the qps levels we deal with here, I feel. Still - don't think this is an anti pattern per se.

Reply:
I wonder how much perf is actually saved by checking for  `isLocalViewBehind(ctx)` outside the synchronized block? We can just do a simple, single synchronized block 

```
synchronized (view) {
        if (isLocalViewBehind(ctx)) {
          HoodieTimeline localTimeline = viewManager.getFileSystemView(basePath).getTimeline();
          LOG.info("Syncing view as client passed last known instant " + lastKnownInstantFromClient
              + " as last known instant but server has the following last instant on timeline :"
              + localTimeline.lastInstant());
          view.sync();
          return true;
          }
          }
         ```

read-write locks are going to add cognitive overhead everytime we read this :)

Reply:
As synced offline, only keeping the synchronized block now.